src/python/arcor2_arserver/rpc/project.py

```diff
@@ -495,13 +536,17 @@ async def update_action_point_using_robot_cb(req: srpc.p.UpdateActionPointUsingR
     assert glob.PROJECT
 
     ap = glob.PROJECT.bare_action_point(req.args.action_point_id)
-    new_pose = await get_end_effector_pose(req.args.robot.robot_id, req.args.robot.end_effector)
+    if not await glob.LOCK.is_write_locked(ap.id, "ui.owner"):
```

Comment:
Tyhle dva řádky se opakují víceméně v každém callbacku týkajícího se akčních bodů - samo o sobě by to tak nevadilo, největší nevýhodu vidím v tom, že tu je definovaná x-krát stejná hláška. Nebylo by lepší udělat něco jako `await ensure_locked(ap.id, ui.owner)`?

Reply:
Případně se na to může udělat celkově nějaká globální funkce s defaultní hláškou, ono se to dost opakuje i ve scéně... A v produkčním režimu už by uživatel vlastně nic takového reálně vidět neměl...

Reply:
Určitě, pokud to bude správně implementováno v editoru, tak toto uživatel nikdy neuvidí. Pro vývojáře editoru by bylo dobré, kdyby ta hláška obsahovala id/jméno čeho se týká.